resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/MultiValuedCollectionParamConverter.java

```diff
@@ -0,0 +1,138 @@
+package org.jboss.resteasy.plugins.providers;
+
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Collection;
+
+import javax.annotation.Priority;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.core.StringParameterInjector;
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+/**
+ * @author Marek Kopecky mkopecky@redhat.com
+ * @author Ron Sigal rsigal@redhat.com
+ */
+@Provider
+@Priority(Integer.MAX_VALUE)
+public class MultiValuedCollectionParamConverter implements ParamConverter<Collection<?>> 
+{
+   private Constructor<?> constructor;
+   private StringParameterInjector stringParameterInjector;
+   private String separator;
+
+   public MultiValuedCollectionParamConverter(Constructor<?> constructor, StringParameterInjector stringParameterInjector, String separator)
+   {
+      this.stringParameterInjector = stringParameterInjector;
+      this.constructor = constructor;
+      this.separator = separator;
+   }
+
+   @Override
+   public String toString(Collection<?> value)
+   {
+      if (value == null)
+      {
+         return null;
+      }
+      return stringify(value);
+   }
+
+   @Override
+   public Collection<?> fromString(String param)
+   {
+      try
+      {
+         Collection<?> c = (Collection<?>) constructor.newInstance();
+         return parse(c, param.split(separator));
+      }
+      catch (Exception e)
+      {
+         throw new RuntimeException(Messages.MESSAGES.unableToParse(param));
+      }
+   }
+
+   ////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+   @SuppressWarnings({ "unchecked", "rawtypes" })
+   private Collection<?> parse(Collection c, String[] params) throws Exception
+   {
+      for (String param : params)
+      {
+         c.add(stringParameterInjector.extractValue(param));
+      }
+      return c;
+   }
+
+   @SuppressWarnings({ "rawtypes", "unchecked" })
+   private String stringify(Collection<?> value)
```

Comment:
You forgot to find and use the `headerDelegate` if any here

Reply:
@NicoNes, yeah, I hadn't gotten that far. It's in there now (but not committed yet).

Reply:
Looks good to me now.